// kloudspot-analytics-node-sdk/lib/AirportResource.js
"use strict";

const URL = require("../config/urls");

class UserInfoResource {
	
	constructor(options) {
		this.apiClient = options.apiClient;
		//console.log('Hierarchy object created -> ', this.apiClient.getToken());
	}
    
    // POST /api/v1/user-info
    bookloads(options, callback) {
      return this.apiClient._postRequest(URL.AIRPORT_RESOURCE.BOOKLOADS, options, callback);
    }
    
    eBoardingStats(options, callback) {
      return this.apiClient._postRequest(URL.AIRPORT_RESOURCE.EBOARDINGSTATS, options, callback);
    }
    
    cussDistribution(options, callback) {
      return this.apiClient._postRequest(URL.AIRPORT_RESOURCE.CUSS_DISTRIBUTION, options, callback);
    }
    
    cuppsDistribution(options, callback) {
      return this.apiClient._postRequest(URL.AIRPORT_RESOURCE.CUPPS_DISTRIBUTION, options, callback);
    }
    
    atrsDistribution(options, callback) {
      return this.apiClient._postRequest(URL.AIRPORT_RESOURCE.ATRS_DISTRIBUTION, options, callback);
    }
    
    aodbAirlineStats(options, callback) {
      return this.apiClient._postRequest(URL.AIRPORT_RESOURCE.AODB_AIRLINE_STATS, options, callback);
    }
    
  }
  
  module.exports = UserInfoResource;
