// kloudspot-analytics-node-sdk/lib/AttributeResource.js
"use strict";

const URL = require("../config/urls");
class AttributeResource {
    constructor(options) {
      this.apiClient = options.apiClient;
    }
  
    // GET /api/v1/attributes
    getAttributes(callback) {
      return this.apiClient._getRequest(URL.ATTRIBUTE_RESOURCE.ATTRIBUTES, callback);
    }
  
    // POST /api/v1/attributes
    postAttribute(options, callback) {
      return this.apiClient._postRequest(URL.ATTRIBUTE_RESOURCE.ATTRIBUTES, options, callback);
    }
  
    // DELETE /api/v1/attributes/{id}
    deleteAttributeById(id, callback) {
      const formattedUrl = URL.ATTRIBUTE_RESOURCE.ATTRIBUTE_BY_ID.replace('{id}', id);
      return this.apiClient._deleteRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/attributes/{id}
    getAttributeById(id, callback) {
      const formattedUrl = URL.ATTRIBUTE_RESOURCE.ATTRIBUTE_BY_ID.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/attributes/{purpose}/{name}
    getAttributeByPurposeAndName(purpose, name, callback) {
      const formattedUrl = URL.ATTRIBUTE_RESOURCE.ATTRIBUTE_BY_PURPOSE_AND_NAME
        .replace('{purpose}', purpose)
        .replace('{name}', name);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  }
  

  module.exports = AttributeResource;
