// kloudspot-analytics-node-sdk/lib/BluetoothAnalytics.js
"use strict";

const URL = require("../config/urls");
class BluetoothAnalytics {
    constructor(options) {
      this.apiClient = options.apiClient;
    }
  
    // GET /api/v1/ble/analytics/deviceSummary
    getBLEDeviceSummary(callback) {
      return this.apiClient._getRequest(URL.BLE_ANALYTICS_RESOURCE.DEVICE_SUMMARY, callback);
    }
  
    // POST /api/v1/ble/analytics/distribution
    postBLEDistribution(options, callback) {
      return this.apiClient._postRequest(URL.BLE_ANALYTICS_RESOURCE.DISTRIBUTION, options, callback);
    }
  
    // GET /api/v1/ble/analytics/generic/list
    getBLEGenericList(callback) {
      return this.apiClient._getRequest(URL.BLE_ANALYTICS_RESOURCE.GENERIC_LIST, callback);
    }
  
    // GET /api/v1/ble/analytics/macsForSite
    getBLEMacsForSite(callback) {
      return this.apiClient._getRequest(URL.BLE_ANALYTICS_RESOURCE.MACS_FOR_SITE, callback);
    }
  
    // GET /api/v1/ble/analytics/sessions/{mac}
    getBLESessionsByMac(mac, callback) {
      const formattedUrl = URL.BLE_ANALYTICS_RESOURCE.SESSIONS_BY_MAC.replace('{mac}', mac);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/ble/analytics/timeline
    getBLETimeline(callback) {
      return this.apiClient._getRequest(URL.BLE_ANALYTICS_RESOURCE.TIMELINE, callback);
    }
  }
  

  module.exports = BluetoothAnalytics;
