// kloudspot-analytics-node-sdk/lib/DeviceAnalyticsResource.js
"use strict";

const URL = require("../config/urls");
class DeviceAnalyticsResource {
    // GET /api/v1/device/analytics/history
    getDeviceAnalyticsHistory(callback) {
      return this.apiClient._getRequest(URL.DEVICE_ANALYTICS_RESOURCE.HISTORY, callback);
    }
  
    // GET /api/v1/device/analytics/locate
    getDeviceAnalyticsLocate(callback) {
      return this.apiClient._getRequest(URL.DEVICE_ANALYTICS_RESOURCE.LOCATE, callback);
    }
  
    // POST /api/v1/device/analytics/navigate
    navigateDeviceAnalytics(options, callback) {
      return this.apiClient._postRequest(URL.DEVICE_ANALYTICS_RESOURCE.NAVIGATE, options, callback);
    }
  
    // POST /api/v1/device/analytics/navigateToZone
    navigateToZoneDeviceAnalytics(options, callback) {
      return this.apiClient._postRequest(URL.DEVICE_ANALYTICS_RESOURCE.NAVIGATE_TO_ZONE, options, callback);
    }
  
    // GET /api/v1/device/analytics/searchPOI
    searchPOIDeviceAnalytics(callback) {
      return this.apiClient._getRequest(URL.DEVICE_ANALYTICS_RESOURCE.SEARCH_POI, callback);
    }
  }
  

  module.exports = DeviceAnalyticsResource;
