// kloudspot-analytics-node-sdk/lib/ExternalKisokResource.js
"use strict";

const URL = require("../config/urls");
class ExternalKioskResource {
    constructor(options) {
      this.apiClient = options.apiClient;
    }
  
    // POST /api/v1/kiosk/checkin
    postKioskCheckin(callback) {
      return this.apiClient._postRequest(URL.EXTERNAL_KIOSK.CHECKIN, callback);
    }
  
    // POST /api/v1/kiosk/checkout
    postKioskCheckout(callback) {
      return this.apiClient._postRequest(URL.EXTERNAL_KIOSK.CHECKOUT, callback);
    }
  
    // GET /api/v1/kiosk/locationMappings
    getKioskLocationMappings(callback) {
      return this.apiClient._getRequest(URL.EXTERNAL_KIOSK.LOCATION_MAPPINGS, callback);
    }
  
    // GET /api/v1/kiosk/userInfo/{login}/{siteId}/{type}
    getKioskUserInfo(login, siteId, type, callback) {
      const formattedUrl = URL.EXTERNAL_KIOSK.USER_INFO.replace('{login}', login)
        .replace('{siteId}', siteId)
        .replace('{type}', type);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  }
  module.exports = ExternalKioskResource;
