// kloudspot-analytics-node-sdk/lib/KloudspotInsights.js
'use strict';

const VERSION = require('../package.json').version;
const AnalyticsResource = require('./AnalyticsResource');
const AnalyticsQueries = require('./AnalyticsResource');
const ApiClient = require('./ApiClient');
const AttributeResources = require('./AttributeResource');
const AuditResource = require('./AuditResource');
const BluetoothAnalytics = require('./BluetoothAnalytics');
const BookingResource = require('./BookingResource');
const CameraMonitoringResource = require('./CameraMonitoringResource');
const ClientDeviceResource = require('./ClientDeviceResource');
const DeviceAnalyticsResource = require('./DeviceAnalyticsResource');
const ExternalKioskResource = require('./ExternalKisokResource');
const HumanAnalyticsResource = require('./HumanAnalyticsResource');
const SiteHierarchy = require('./SiteHierarchy');
const VisionAnalytics = require('./VisionAnalytics');
const UserInfoResource = require('./UserInfoResource');
const AirportResource = require('./AirportResource');
const MLResource = require('./MLResource');
const IOTAnalyticsResource = require('./IOTAnalyticsResource');
const QueueResource = require('./QueueResource');

class KloudspotInsights {

	constructor(options) {

		var that = this;

		if (!(this instanceof KloudspotInsights)) {
			return new KloudspotInsights(options);
		}

		this.VERSION = VERSION;

		that.apiClient = new ApiClient(options);
		this.siteHierarchy = new SiteHierarchy({ 'apiClient': this.apiClient });
		this.visionAnalytics = new VisionAnalytics({ 'apiClient': this.apiClient });
		this.analyticsResource = new AnalyticsResource({ 'apiClient': this.apiClient })
		this.attributeResource = new AttributeResources({ 'apiClient': this.apiClient });
		this.auditResource = new AuditResource({ 'apiClient': this.apiClient });
		this.bluetoothAnalytics = new BluetoothAnalytics({ 'apiClient': this.apiClient });
		this.bookingResources = new BookingResource({ 'apiClient': this.apiClient });
		this.cameraMonitoringResource = new CameraMonitoringResource({ 'apiClient': this.apiClient });
		this.clientDeviceResource = new ClientDeviceResource({ 'apiClient': this.apiClient });
		this.deviceAnalyticsResource = new DeviceAnalyticsResource({ 'apiClient': this.apiClient });
		this.externalKiskoResource = new ExternalKioskResource({ 'apiClient': this.apiClient });
		this.humanAnalyticsResource = new AnalyticsQueries({ 'apiClient': this.apiClient });
		this.humanAnalyticsResource = new HumanAnalyticsResource({ 'apiClient': this.apiClient });
		this.userResource = new UserInfoResource({ 'apiClient': this.apiClient });
		this.airport = new AirportResource({ 'apiClient': this.apiClient });
		this.machinelearning = new MLResource({ 'apiClient': this.apiClient });
		this.iot = new IOTAnalyticsResource({ 'apiClient': this.apiClient });
		this.queue = new QueueResource({ 'apiClient': this.apiClient });
		// this.analyticsQueries = new AnalyticsQueries({'apiClient':this.apiClient});

	}

}

module.exports = KloudspotInsights;
