// kloudspot-analytics-node-sdk/lib/MonitoringResource.js
"use strict";

const URL = require("../config/urls");
class MonitoringResource {
    // GET /api/v1/system-status
    getSystemStatus(callback) {
      return this.apiClient._getRequest(URL.MONITORING_RESOURCE.SYSTEM_STATUS, callback);
    }
  }
  
  class TagResource {
    // GET /api/v1/tags
    getTags(callback) {
      return this.apiClient._getRequest(URL.MONITORING_RESOURCE.TAGS, callback);
    }
  
    // POST /api/v1/tags
    createTag(options, callback) {
      return this.apiClient._postRequest(URL.MONITORING_RESOURCE.TAGS, options, callback);
    }
  
    // DELETE /api/v1/tags/{id}
    deleteTagById(id, callback) {
      const formattedUrl = URL.MONITORING_RESOURCE.TAG_BY_ID.replace('{id}', id);
      return this.apiClient._deleteRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/tags/{id}
    getTagById(id, callback) {
      const formattedUrl = URL.MONITORING_RESOURCE.TAG_BY_ID.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/tags/{id}/image
    getTagImageById(id, callback) {
      const formattedUrl = URL.MONITORING_RESOURCE.IMAGE_BY_ID.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // POST /api/v1/tags/{id}/image
    setTagImageById(id, options, callback) {
      const formattedUrl = URL.MONITORING_RESOURCE.IMAGE_BY_ID.replace('{id}', id);
      return this.apiClient._postRequest(formattedUrl, options, callback);
    }
  }
  

  module.exports = MonitoringResource;
