// kloudspot-analytics-node-sdk/lib/NotificationResource.js
"use strict";

const URL = require("../config/urls");
class NotificationHistoryResource {
    // DELETE /api/v1/notificationHistory
    deleteNotificationHistory(callback) {
      return this.apiClient._deleteRequest(
        URL.NOTIFICATION_HISTORY_RESOURCE.NOTIFICATION_HISTORY,
        callback
      );
    }
  
    // GET /api/v1/notificationHistory
    getNotificationHistory(callback) {
      return this.apiClient._getRequest(
        URL.NOTIFICATION_HISTORY_RESOURCE.NOTIFICATION_HISTORY,
        callback
      );
    }
  
    // POST /api/v1/notificationHistory/notificationSearch
    searchNotificationHistory(options, callback) {
      return this.apiClient._postRequest(
        URL.NOTIFICATION_HISTORY_RESOURCE.NOTIFICATION_HISTORY_SEARCH,
        options,
        callback
      );
    }
  
    // GET /api/v1/notificationHistory/page
    getNotificationHistoryPage(callback) {
      return this.apiClient._getRequest(
        URL.NOTIFICATION_HISTORY_RESOURCE.NOTIFICATION_HISTORY_PAGE,
        callback
      );
    }
  
    // DELETE /api/v1/notificationHistory/{id}
    deleteNotificationById(notificationId, callback) {
      
      return this.apiClient._deleteRequest(URL.NOTIFICATION_HISTORY_RESOURCE.NOTIFICATION_HISTORY+'/'+notificationId, callback);
    }
  
    // GET /api/v1/notificationHistory/{id}
    getNotificationById(notificationId, callback) {
      
      return this.apiClient._getRequest(URL.NOTIFICATION_HISTORY_RESOURCE.NOTIFICATION_HISTORY+'/'+notificationId, callback);
    }
  }
  

  module.exports = NotificationHistoryResource;
