// kloudspot-analytics-node-sdk/lib/SettingResource.js
"use strict";

const URL = require("../config/urls");
class SettingResource {
    // POST /api/v1/setting/change-retention
    changeRetention(options, callback) {
      return this.apiClient._postRequest(URL.SETTING_RESOURCE.CHANGE_RETENTION, options, callback);
    }
  
    // POST /api/v1/setting/ssl/upload
    sslUpload(options, callback) {
      return this.apiClient._postRequest(URL.SETTING_RESOURCE.SSL_UPLOAD, options, callback);
    }
  
    // GET /api/v1/setting/system/ext-auth
    getSystemExternalAuth(callback) {
      return this.apiClient._getRequest(URL.SETTING_RESOURCE.SYSTEM_EXT_AUTH, callback);
    }
  
    // POST /api/v1/setting/system/ext-auth
    setSystemExternalAuth(options, callback) {
      return this.apiClient._postRequest(URL.SETTING_RESOURCE.SYSTEM_EXT_AUTH, options, callback);
    }
  
    // GET /api/v1/setting/system/hybrid-settings-csv
    getSystemHybridSettingsCsv(callback) {
      return this.apiClient._getRequest(URL.SETTING_RESOURCE.SYSTEM_HYBRID_SETTINGS_CSV, callback);
    }
  
    // GET /api/v1/setting/system/timezones
    getSystemTimezones(callback) {
      return this.apiClient._getRequest(URL.SETTING_RESOURCE.SYSTEM_TIMEZONES, callback);
    }
  
    // DELETE /api/v1/setting/system/{id}
    deleteSystemById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.SYSTEM_BY_ID.replace('{id}', id);
      return this.apiClient._deleteRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/setting/system/{id}
    getSystemById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.SYSTEM_BY_ID.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // POST /api/v1/setting/system/{id}
    setSystemById(id, options, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.SYSTEM_BY_ID.replace('{id}', id);
      return this.apiClient._postRequest(formattedUrl, options, callback);
    }
  
    // GET /api/v1/setting/system/{id}/image
    getSystemImageById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.SYSTEM_BY_ID_IMAGE.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // POST /api/v1/setting/system/{id}/image
    setSystemImageById(id, options, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.SYSTEM_BY_ID_IMAGE.replace('{id}', id);
      return this.apiClient._postRequest(formattedUrl, options, callback);
    }
  
    // GET /api/v1/setting/system/{id}/image-scaled
    getSystemImageScaledById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.SYSTEM_BY_ID_IMAGE_SCALED.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // DELETE /api/v1/setting/user/{id}
    deleteUserById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.USER_BY_ID.replace('{id}', id);
      return this.apiClient._deleteRequest(formattedUrl, callback);
    }
  
    // GET /api/v1/setting/user/{id}
    getUserById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.USER_BY_ID.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // POST /api/v1/setting/user/{id}
    setUserById(id, options, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.USER_BY_ID.replace('{id}', id);
      return this.apiClient._postRequest(formattedUrl, options, callback);
    }
  
    // GET /api/v1/setting/user/{id}/image
    getUserImageById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.USER_BY_ID_IMAGE.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  
    // POST /api/v1/setting/user/{id}/image
    setUserImageById(id, options, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.USER_BY_ID_IMAGE.replace('{id}', id);
      return this.apiClient._postRequest(formattedUrl, options, callback);
    }
  
    // GET /api/v1/setting/user/{id}/image-scaled
    getUserImageScaledById(id, callback) {
      const formattedUrl = URL.SETTING_RESOURCE.USER_BY_ID_IMAGE_SCALED.replace('{id}', id);
      return this.apiClient._getRequest(formattedUrl, callback);
    }
  }
  
  module.exports = SettingResource;
