// kloudspot-analytics-node-sdk/config/urls.js
module.exports = {
  LOGIN: "/api/v1/auth/login",
  SITES: "/api/v1/locations/sites",
  HIERARCHY: "/api/v1/locations/hierarchy",
  LOCATE: "/api/v1/device/analytics/locate",
  SEARCH: "/api/v1/analytics/deviceSearch",
  FIND: "/api/v1/analytics/devicesForSite?deviceTypes=probe&deviceTypes=bluetooth&excludePasserBy=false",
  APSTATUS: "/api/v1/system-status",
  DISTRIBUTION: "/api/v1/analytics/distribution",

  //List all site hierarchy apis here
  HIERARCHY: {
    SITES: "/api/v1/locations/sites",
    LOCATIONS: "/api/v1/locations/locations",
    ALLTAGS: "/api/v1/locations/all-tags",
    ACCESSPOINT: "/api/v1/locations/ap",
    CSVUPLOADSAMPLE: "/api/v1/locations/csv-upload-sample",
    FLOORS: "/api/v1/locations/floors",
    GROUPTAGINFOSPOTID: "/api/v1/locations/groupTags/effective/infopot/",
    GROUPTAGLOCATIONID: "/api/v1/locations/groupTags/effective/location/",
    LOCATIONHIERARHCY: "/api/v1/locations/hierarchy",
    INFOSPOTSINZONES: "/api/v1/locations/infospot/{id}/zones",
    INFOSPOTSBYLOGIN: "/api/v1/locations/infospots/",
    ADD: "/api/v1/locations/location",
    UPDATECHILD: "/api/v1/locations/location/updateChild/{id}",
    DELETE: "/api/v1/locations/location/{id}",
    LOCATIONSUMMARY: "/api/v1/locations/locationSummary",
    MINIHIERARCHY: "/api/v1/locations/miniHierarchy",
    MONITORINGOFF: "/api/v1/locations/monitoring-off",
    MONITORINGOFFID: "/api/v1/locations/monitoring-off/{id}",
    MONITORINGON: "/api/v1/locations/monitoring-on",
    MONITORINGONID: "/api/v1/locations/monitoring-on/{id}",
    MYINFOSPOTS: "/api/v1/locations/myInfospots",
    NONLEAVES: "/api/v1/locations/nonleaves",
    NONLEAVESSUMMARY: "/api/v1/locations/nonleavesSummary",
    SHORTESTPATH: "/api/v1/locations/shortestPath",
    UPLOAD: "/api/v1/locations/upload",
    ZONEINFOSPOTS: "/api/v1/locations/zone/{id}/infospots",
    LOCATIONBYID: "/api/v1/locations/{id}",
    BREADCRUMBS: "/api/v1/locations/{id}/breadcrumbs",
    LOCATIONBYIDCSV: "/api/v1/locations/{id}/csv",
    SITEBYID: "/api/v1/locations/{id}/sites",
    SITESBYID: "/api/v1/locations/{id}/sitesS",
    ZONEBYID: "/api/v1/locations/{id}/zones",
    ZONESBYID: "/api/v1/locations/{id}/zonesS",
  },
  //List all Vision Analytics apis here
  VISION: {
    DISTRIBUTION: "/api/v1/camera/analytics/distribution",
    DWELL: "/api/v1/camera/analytics/dwell",
    DWELL_DISTRIBUTION: "/api/v1/camera/analytics/dwellDistribution",
    ENTITY_DEMOGRAPHICS: "/api/v1/camera/analytics/entityDemographics",
    ENTRY_EXIT: "/api/v1/camera/analytics/entryExit",
    MAX_DWELL_TIME: "/api/v1/camera/analytics/maxDwellTime",
    MOVEMENT_TREE: "/api/v1/camera/analytics/movementTree",
    MOVEMENTS: "/api/v1/camera/analytics/movements",
    PAIRWISE_MOVEMENT: "/api/v1/camera/analytics/pairwiseMovement",
  },
  HUMAN_IMAGES: {
    ENTITY_UPLOAD: "/api/v1/tusUpload/entity-upload",
    ENTITY_UPLOAD_SEARCH: "/api/v1/tusUpload/search",
    ENTITY_UPLOAD_SEARCH_BY_CAMERA: "/api/v1/tusUpload/searchByCamera",
    TUS_UPLOAD_REGISTRATION: "/api/v1/tusUpload/registration",
  },
  ANALYTICS_QUERY: {
    CLIENT_MOVEMENTS: "/api/v1/analytics/clientMovements",
    DAILY_PRESENCE: "/api/v1/analytics/dailyPresence/{mac}",
    DEVICE_DEMOGRAPHICS: "/api/v1/analytics/deviceDemographics",
    DEVICE_SEARCH: "/api/v1/analytics/deviceSearch",
    DEVICES_FOR_SITE: "/api/v1/analytics/devicesForSite",
    DISTRIBUTION: "/api/v1/analytics/distribution",
    GENERIC: "/api/v1/analytics/generic",
    GENERIC_LIST: "/api/v1/analytics/generic/list",
    MOVEMENT_TREE: "/api/v1/analytics/movementTree",
    PAIRWISE_MOVEMENT: "/api/v1/analytics/pairwiseMovement",
    REPEATED_USERS: "/api/v1/analytics/repeatedUsers",
    RUN_QUERY: "/api/v1/analytics/runQuery",
    SUMMARY: "/api/v1/analytics/summary",
    TIMELINE: "/api/v1/analytics/timeline",
    TZINFO: "/api/v1/analytics/tzinfo",
    USER_DETAILS: "/api/v1/analytics/userdetails",
    USER_DETAILS_BY_MAC: "/api/v1/analytics/userdetails/{mac}",
  },
  IMAGE_MANAGEMENT: {
    IMAGES: "/api/v1/images",
    IMAGE_BY_ID: "/api/v1/images/image/{id}",
    IMAGE_BY_ID_DOWNSCALED: "/api/v1/images/image/{id}/downscaled",
    IMAGE_TILES: "/api/v1/images/tiles/{id}/{z}/{x}/{y}",
    IMPORT_IMAGE: "/api/v1/images/{id}/import",
    UPLOAD_IMAGE: "/api/v1/images/{id}/upload",
  },

  IOT_ANALYTICS_QUERY: {
    DISTRIBUTION: "/api/v1/iot/analytics/distribution",
    GRID_DISTRIBUTION: "/api/v1/iot/analytics/gridDistribution",
  },
  ATTRIBUTE_RESOURCE: {
    ATTRIBUTES: "/api/v1/attributes",
    ATTRIBUTE_BY_ID: "/api/v1/attributes/{id}",
    ATTRIBUTE_BY_PURPOSE_AND_NAME: "/api/v1/attributes/{purpose}/{name}",
  },
  AUDIT_RESOURCE: "/api/v1/audits",

  BLE_ANALYTICS_RESOURCE: {
    DEVICE_SUMMARY: "/api/v1/ble/analytics/deviceSummary",
    DISTRIBUTION: "/api/v1/ble/analytics/distribution",
    GENERIC_LIST: "/api/v1/ble/analytics/generic/list",
    MACS_FOR_SITE: "/api/v1/ble/analytics/macsForSite",
    SESSIONS_BY_MAC: "/api/v1/ble/analytics/sessions/{mac}",
    TIMELINE: "/api/v1/ble/analytics/timeline",
  },
  CAMERA_MONITORING_RESOURCE: {
    CAMERA_STATUS: "/api/v1/camera-status",
  },
  DEVICE_ANALYTICS_RESOURCE: {
    HISTORY: "/api/v1/device/analytics/history",
    LOCATE: "/api/v1/device/analytics/locate",
    NAVIGATE: "/api/v1/device/analytics/navigate",
    NAVIGATE_TO_ZONE: "/api/v1/device/analytics/navigateToZone",
    SEARCH_POI: "/api/v1/device/analytics/searchPOI",
  },
  CLIENT_DEVICE_RESOURCE: {
    DEVICES: "/api/v1/devices",
    BULK: "/api/v1/devices/bulk",
    CSV_IMPORT: "/api/v1/devices/csv-import",
    EMAIL: "/api/v1/devices/email",
    LINK_CSV_IMPORT: "/api/v1/devices/ezLinkCSVImport",
    IDENTIFIED: "/api/v1/devices/identified",
    OUI: "/api/v1/devices/oui",
    TAGS: "/api/v1/devices/tags",
    BY_ID: "/api/v1/devices/{id}",
    DEVICES_CSV_EXPORT: "/api/v1/devices/csv-export",
    DEVICE_HISTORY_BY_ID: "/api/v1/devices/history/{id}",
  },
  IOT_CAMERA_ANALYTICS: {
    IOT_LOCATIONS: "/api/v1/iotAnalytics/iot-locations",
    IOT_LOCATIONS_GRIDS: "/api/v1/iotAnalytics/iot-locations/grids",
  },
  LOCATION_ANALYTICS_RESOURCE: {
    DENSITY_AROUND_INDEX: "/api/v1/location/analytics/densityAroundIndex",
    DISTRIBUTION: "/api/v1/location/analytics/distribution",
    FLOOR_DENSITIES: "/api/v1/location/analytics/floorDensities",
  },
  LOCATION_ATTRIBUTES_RESOURCE: {
    LOCATION_ATTRIBUTES: "/api/v1/locationAttributes",
    LOCATION_ATTRIBUTE_BY_ID: "/api/v1/locationAttributes/{id}",
  },
  NETWORK_ELEMENT_EXTERNAL_RESOURCE: {
    NETWORK_ELEMENTS_BY_LOCATION: "/api/v1/networkElements/location/{locationId}",
    UPDATE_ACCESS_POINTS: "/api/v1/networkElements/update/accessPoints",
    UPDATE_CAMERAS: "/api/v1/networkElements/update/cameras",
  },
  NOTIFICATION_HISTORY_RESOURCE: {
    NOTIFICATION_HISTORY: "/api/v1/notificationHistory",
    NOTIFICATION_HISTORY_PAGE: "/api/v1/notificationHistory/page",
    NOTIFICATION_HISTORY_SEARCH: "/api/v1/notificationHistory/notificationSearch",
  },
  REPORT_RESOURCE: {
    CSV_REPORT: "/api/v1/reports/csvReport/{reportName}",
    CSV_REPORT_BY_QUERY: "/api/v1/reports/csvReport/{reportName}/{queryName}",
  },
  SETTING_RESOURCE: {
    CHANGE_RETENTION: "/api/v1/setting/change-retention",
    SSL_UPLOAD: "/api/v1/setting/ssl/upload",
    SYSTEM_EXT_AUTH: "/api/v1/setting/system/ext-auth",
    SYSTEM_HYBRID_SETTINGS_CSV: "/api/v1/setting/system/hybrid-settings-csv",
    SYSTEM_TIMEZONES: "/api/v1/setting/system/timezones",
    SYSTEM_BY_ID: "/api/v1/setting/system/{id}",
    SYSTEM_BY_ID_IMAGE: "/api/v1/setting/system/{id}/image",
    SYSTEM_BY_ID_IMAGE_SCALED: "/api/v1/setting/system/{id}/image-scaled",
    USER_BY_ID: "/api/v1/setting/user/{id}",
    USER_BY_ID_IMAGE: "/api/v1/setting/user/{id}/image",
    USER_BY_ID_IMAGE_SCALED: "/api/v1/setting/user/{id}/image-scaled",
  },
  MONITORING_RESOURCE: {
    SYSTEM_STATUS: "/api/v1/system-status",
    TAGS: "/api/v1/tags",
    TAG_BY_ID: "/api/v1/tags/{id}",
    IMAGE_BY_ID: " /api/v1/tags/{id}/image"
  },
  TRANSACTION_ANALYTICS_RESOURCE: {
    AGGR_TRANSACTIONS: "/api/v1/transactions/analytics/aggrTransactions",
    BILLS_COUNT: "/api/v1/transactions/analytics/-bills-count",
    TRANSACTIONS: "/api/v1/transactions/analytics/transactions",
    UPLOAD_EXCEL: "/api/v1/transactions/analytics/upload-excel",
  },
  USER_INFO_RESOURCE: {
    USER_INFO: "/api/v1/user-info",
    USER_SEARCH: "/api/v1/user-info/search",
  },
  USER_DEVICE_INFO_RESOURCE: {
    USER_DEVICE: "/api/v1/userDevice",
  },
  USER_MANAGEMENT_RESOURCE: {
    USERS: "/api/v1/users",
  },
  VISION_ENTITY_RESOURCE: {
    FINGERPRINT: "/api/v1/vision/fingerprint",
  },
  HUMANS: {
    HUMANS: "/api/v1/humans",
    ALLHUMANIMAGES: "/api/v1/humans/allHumanImages",
    BULKSTATUSUPDATE: "/api/v1/humans/bulkStatusUpdate",
    DELETE: "/api/v1/humans/delete",
    FINDBYUNIQUEID: "/api/v1/humans/findByUniqueId/{uniqueId}",
    GET: "/api/v1/humans/get",
    GETALL: "/api/v1/humans/getAll",
    GETUNREGISTEREDHUMANLIST: "/api/v1/humans/getUnregisteredHumanList",
    HUMANRECORDS: "/api/v1/humans/human-records",
    UPLOADMETADATA: "/api/v1/humans/upload-metadata",
    ZIP: "/api/v1/humans/zip",
    CREATEINDEX: "/api/v1/humans/{indexName}/createIndex",
  },
  BOOKING: {
    BOOKING: "/api/v1/booking",
    BOOKING_CHECKOUT: "/api/v1/booking/checkout",
    BOOKING_CLAIM: "/api/v1/booking/claim",
    BOOKING_CURRENT: "/api/v1/booking/current",
    BOOKING_DELEGATED: "/api/v1/booking/delegated",
    BOOKING_DELEGATEE: "/api/v1/booking/delegatee",
    BOOKING_DELEGATEE_MODIFY: "/api/v1/booking/delegatee/modify",
    BOOKING_DELEGATEE_PAGEABLE: "/api/v1/booking/delegatee/pageable",
    BOOKING_DELETE_FUTURE: "/api/v1/booking/deleteFuture/{id}/{subid}",
    BOOKING_DIRECT_REPORTS: "/api/v1/booking/directreports",
    BOOKING_EXTEND: "/api/v1/booking/extend",
    BOOKING_INFOSPOT: "/api/v1/booking/infospot/{infospotId}",
    BOOKING_ISSUE: "/api/v1/booking/issue",
    BOOKING_LOCATION: "/api/v1/booking/location/{id}",
    BOOKING_MODIFY: "/api/v1/booking/modify",
    BOOKING_RE_ENABLE: "/api/v1/booking/re-enable",
    BOOKING_SHARE: "/api/v1/booking/share",
    BOOKING_USER_PAGEABLE: "/api/v1/booking/user/pageable/{login}",
    BOOKING_USER: "/api/v1/booking/user/{login}",
    BOOKING_USER_CIRCLE: "/api/v1/booking/usercircle",
    BOOKING_ZONE: "/api/v1/booking/zone/{id}",
    BOOKING_BY_ID: "/api/v1/booking/{id}",
    BOOKING_BY_ID_SUBID: "/api/v1/booking/{id}/{subid}"

  },
  EXTERNAL_KIOSK: {
    CHECKIN: "/api/v1/kiosk/checkin",
    CHECKOUT: "/api/v1/kiosk/checkout",
    LOCATION_MAPPINGS: "/api/v1/kiosk/locationMappings",
    USER_INFO: "/api/v1/kiosk/userInfo/{login}/{siteId}/{type}"
  },
  ML_PREDICTION: {
    PREDICT: "/api/v1/models/predict"
  },
  TASK: {
    SEARCH: "/api/v1/tasks/search"

  },
  SYMPHONICT: {
    ALL_OPENSPACES: "/api/v1/symphonict/all-openspaces",
    ALL_PROJECTS: "/api/v1/symphonict/all-projects",
    MESSAGES: "/api/v1/symphonict/messages",
    USER_INFO: "/api/v1/symphonict/user-info",
    ZOOM_MTG_INFO: "/api/v1/symphonict/zoom-mtg-info"
  },
  AIRPORT_RESOURCE: {
	  BOOKLOADS: "/api/v1/airport/bookloads",
	  EBOARDINGSTATS: "/api/v1/airport/eboardingStats",
	  ATRS_DISTRIBUTION: "/api/v1/airport/atrs/distribution",
	  CUPPS_DISTRIBUTION: "/api/v1/airport/cupps/distribution",
	  CUSS_DISTRIBUTION: "/api/v1/airport/cuss/distribution",
	  AODB_AIRLINE_STATS: "/api/v1/airport/aodb/airline-stats"
  },
  ML_RESOURCE: {
	  PREDICT: "/api/v1/models/predict"
  },
  QUEUE: {
	  LOAD: "/api/v1/queue/get",
	  DISTRIBUTION: "/api/v1/queue/distribution",
	  MAX_WAITTIME: "/api/v1/queue/maxWaitTime"
  },
};
